app/src/main/java/com/android/streetworkapp/ui/navigation/NavigationActions.kt

```diff
@@ -0,0 +1,85 @@
+package com.android.streetworkapp.ui.navigation
+
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.outlined.AccountCircle
+import androidx.compose.material.icons.outlined.Place
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.navigation.NavHostController
+
+object Route {
+  const val AUTH = "Auth"
+  const val MAP = "Map"
+  const val PROFILE = "Profile"
+  const val UNK = "TBD" // TODO: not yet defined
+}
+
+object Screen {
+  const val AUTH = "Auth Screen"
+  const val MAP = "Map Screen"
+  const val PROFILE = "Profile Screen"
+  const val UNK = "TBD Screen" // TODO: not yet defined
+}
+
+data class TopLevelDestination(val route: String, val icon: ImageVector, val textId: String)
+
+// all data classes here are reachable through user actions (i.e not auth)
+object TopLevelDestinations {
+  val MAP = TopLevelDestination(route = Route.MAP, icon = Icons.Outlined.Place, textId = "Map")
+  val PROFILE =
+      TopLevelDestination(
+          route = Route.PROFILE, icon = Icons.Outlined.AccountCircle, textId = "Profile")
+}
+
+val LIST_TOP_LEVEL_DESTINATION = listOf(TopLevelDestinations.MAP, TopLevelDestinations.PROFILE)
+
+open class NavigationActions(
+    private val navController: NavHostController,
+) {
+  /**
+   * Navigate to the specified [TopLevelDestination]
+   *
+   * @param destination The top level destination to navigate to.
+   *
+   * Clear the back stack when navigating to a new destination.
+   */
+  open fun navigateTo(
```

Comment:
It is possible for this function to fail (like a wrong navigation)? If this is the case, shouldn't the function should log something?